arangod/Aql/ShortestPathNode.cpp

```diff
@@ -142,9 +142,9 @@ ShortestPathNode::ShortestPathNode(ExecutionPlan* plan, size_t id, TRI_vocbase_t
 
 /// @brief Internal constructor to clone the node.
 ShortestPathNode::ShortestPathNode(
-    ExecutionPlan* plan, size_t id, TRI_vocbase_t* vocbase,
-    std::vector<std::unique_ptr<Collection>> const& edgeColls,
-    std::vector<std::unique_ptr<Collection>> const& vertexColls,
+    ExecutionPlan* plan, ExecutionNodeId id, TRI_vocbase_t* vocbase,
+    std::vector<Collection*> const& edgeColls,
+    std::vector<Collection*> const& vertexColls,
```

Comment:
nanu, away from smart pointers?

Reply:
Generally, all `aql::Collections` are held centrally by the `Query`. Previously, the graph nodes built their own objects; which was confusing, because due to that there could be multiple different objects for the same collection.
I removed that and instead used the central Query Collections registry, and thus had to remove the smart pointers.